index.js

```diff
@@ -77,10 +77,31 @@ function createHtmlParser() {
   var Parser = function () {};
 
   Parser.prototype.parseFromString = function (string) {
-    var newDoc = _document.implementation.createHTMLDocument('');
+    var newDoc;
+    if (_document.implementation.createHTMLDocument) {
+      newDoc = _document.implementation.createHTMLDocument('');
+    }
+    else {
+      // IE8 Workaround originally courtesy of https://github.com/Reffyy
+      newDoc = _document.createElement('body');
+      newDoc.documentElement = newDoc;
+      newDoc.body = newDoc;
+    }
 
     if (string.toLowerCase().indexOf('<!doctype') > -1) {
-      newDoc.documentElement.innerHTML = string;
+      try {
+        newDoc.documentElement.innerHTML = string;
+      }
+      catch (e) {
+        // IE9 doesn't allow setting innerHTML on the document, and because IE9 strips <body>
+        // tags going into another body/div we will attempt to extract the body manually
+        var matches = string.match(/<body([^>]*)?>[\s\S]*<\/body>/im);
```

Comment:
I won't be able to test until sometime during next week, but reading the code I don't see how it would work in IE9. Perhaps I am missing something.

There are comments around a solution for IE9 in the linked file, but with `<!doctype` provided it appears to make the same call which I have needed to wrap in a try block. When I tested, calling `doc.documentElement.innerHTML = markup` would throw an exception in IE9.

I will give it a go.